inst/include/dust/dust_cpu.hpp

```diff
@@ -403,21 +403,20 @@ class dust_cpu {
       throw std::invalid_argument(msg.str());
     }
 
-    if (particles_.size() == n_particles) {
+    if (particles_.empty()) {
+      particles_.reserve(n_particles);
+      for (size_t i = 0; i < n_particles; ++i) {
+        particles_.push_back(p);
+      }
```

Comment:
Is this method or `.resize(n)` followed by `vec[i] = init` faster? Are they the same? I've often wondered but don't know the answer

Reply:
resize relies on the thing being stored being default constructible - it will be constructed to that length.

As to faster:

* if resize is possible we have the cost of default-constructing the particle, followed by benefit of subsequently being able to construct them in parallel
* reserve/push wastes no time on default construction but can't be done in parallel